backend/src/main/java/com/example/backend/controller/GoalsController.java

```diff
@@ -68,4 +70,84 @@ public ResponseEntity<?> updateGoal(@PathVariable Long id, @RequestParam String
         return ResponseEntity.ok(goal.getGoalName());
     }
 
+
+    private void checkTicks(Goal g){
+        List<GoalElement> subgoals = goalElementRepository.findGoalElementsByGoal(g);
+        boolean flag = true;
+        for (GoalElement subgoal: subgoals){
+            if(! subgoal.isAchieved()){
+                flag = false;
+                break;
+            }
+        }
+
+        // tu po ukońcczeniu ostatniego podcelu wywalamy cel z bazy
+        // troszkę kontrowersyjne, ale powinno działać
```

Comment:
Ale dlaczego? Przecież jest podział na cele ukończone i nieukończone. Jak cel jest w całości odznaczony, to po prostu jest niżej na wyświetlanej liście.

Reply:
Cel można ręcznie usunąć (klikając na kosz obok niego) - wydaje mi się, że takie podejście jest dużo bardziej intuicyjne.